app/services/scrape.py

```diff
@@ -218,3 +235,62 @@ def get_frequency(self, user_id: str) -> Tuple[TimeSeries, TimeSeries]:
             value=hour_of_day.tolist(),
         )
         return dow_resp, hod_resp
+
+    def get_ml_user(self, username: str):
+        user_fields = [
+            "created_at",
+            "default_profile",
+            "default_profile_image",
+            "description",
+            "favourites_count",
+            "followers_count",
+            "friends_count",
+            "listed_count",
+            "name",
+            "protected",
+            "screen_name",
+            "statuses_count",
+            "verified",
+        ]
+        user_api = self.api.get_user(screen_name=username)
+        user = {"crawled_at": datetime.utcnow()}
+        for field in user_fields:
+            user[field] = getattr(user_api, field)
+        return user
+
+    def _get_conversation(self, conversation_id: int):
+        return list(
+            tweepy.Paginator(
+                self.api_v2.search_recent_tweets,
+                query=f"conversation_id:{conversation_id}",
+                tweet_fields="referenced_tweets",
+            ).flatten()
+        )
+
+    def _get_retweets(self, tweet_id: int):
+        retweets = self.api.get_retweets(tweet_id, trim_user=True)
+        return [
+            {"id": tweet.id, "text": tweet.text, "parent_id": tweet_id}
+            for tweet in retweets
+        ]
+
+    def get_ml_tweets(self, user_id: Union[int, str]):
+        tweets = []
+        duration = timedelta(days=7)
+        last_week = datetime.utcnow() - duration
+        for tweet in tweepy.Paginator(
+            self.api_v2.get_users_tweets,
+            id=user_id,
+            tweet_fields="conversation_id",
+            start_time=last_week.strftime("%Y-%m-%dT%H:%M:%SZ"),
+            max_results=100,
+        ).flatten():
+            tweets.append({"id": tweet.id, "text": tweet.text, "parent_id": 0})
+
+            if not tweet.text.startswith("RT @"):
```

Comment:
- `RT @` Means `referenced tweet` right? how do you know?
- I dint see our docs specify that we exclude `retweets` from user.